libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/HttpClient.kt

```diff
@@ -0,0 +1,65 @@
+package com.mapbox.services.android.navigation.v5.internal.navigation
+
+import com.mapbox.navigator.HttpCode
+import com.mapbox.navigator.HttpInterface
+import com.mapbox.navigator.HttpResponse
+import com.mapbox.services.android.navigation.BuildConfig
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.logging.HttpLoggingInterceptor
+import okio.Okio
+import timber.log.Timber
+import java.io.ByteArrayOutputStream
+
+class HttpClient(
+    private val userAgent: String,
```

Comment:
```suggestion
    private val userAgent: String?,
```
Sorry for misunderstanding. I mean we could make `userAgent` nullable and if it's `null` we will noted this issue. But if it was just empty string it would be harder to note it. wdyt?

Reply:
we can change it to nullable and then add a header only if != null
userAgent?.let{
    addHeader("User-Agent", it)
}
wdyt?

Reply:
or maybe we can make it non null and check
if (userAgent.isEmpty()) {
    throw RuntimeException(message)
}

Reply:
maybe exception is better

Reply:
discussed. will add a `nonEmptyCheck` with exception